integration-cli/docker_cli_build_test.go

```diff
@@ -4694,3 +4694,92 @@ func TestBuildDockerfileOutsideContext(t *testing.T) {
 
 	logDone("build - Dockerfile outside context")
 }
+
+func TestBuildSpaces(t *testing.T) {
+	// Test to make sure that leading/trailing spaces on a command
+	// doesn't change the error msg we get
+	var (
+		err1 error
+		err2 error
+	)
+
+	name := "testspaces"
+	defer deleteImages(name)
+	ctx, err := fakeContext("FROM busybox\nRUN\n",
+		map[string]string{
+			"Dockerfile": "FROM busybox\nRUN\n",
+		})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Close()
+
+	timeRE := regexp.MustCompile(`time="[^Z]*Z"`)
+
+	if _, err1 = buildImageFromContext(name, ctx, false); err1 == nil {
+		t.Fatal("Build 1 was supposed to fail, but didn't")
+	}
+
+	ctx.Add("Dockerfile", "FROM busybox\nRUN    ")
+	if _, err2 = buildImageFromContext(name, ctx, false); err2 == nil {
+		t.Fatal("Build 2 was supposed to fail, but didn't")
+	}
+
+	// Mask the times
+	e1 := timeRE.ReplaceAllString(err1.Error(), "time")
+	e2 := timeRE.ReplaceAllString(err2.Error(), "time")
+
+	if strings.Replace(e1, " ", "", -1) != strings.Replace(e2, " ", "", -1) {
```

Comment:
as agreed with @duglin, it would be better to not "remove the time part" of the error message, but instead use `strings.Index(err.Error(), `level="`)` and do the err comparison from there.